web/js/codeworld_shared.js

```diff
@@ -402,12 +402,13 @@ function getQualifierPrefix(cm, pos) {
   return prefix;
 }
 
-function substitutionCost(a, b, fixedLen) {
+function substitutionCost(a, b, fixedLen, isTermReplaced) {
   const insertCost = 1;
   const deleteCost = 1.5;
   const transCost = 1;
   const substCost = 1.5;
   const caseCost = 0.1;
+  const penalty = 5;
```

Comment:
Let's call this `redirectPenalty` so it's clearer what it means without reading the code later.